lms/djangoapps/courseware/views.py

```diff
@@ -862,6 +873,60 @@ def get_cosmetic_display_price(course, registration_price):
         return _('Free')
 
 
+class EnrollStaffView(View):
+    """
+        Determine If user is global staff, it will be redirected to page "enroll_satff.html"
+        that asks you if you want to register for the course.
+        This pages has the courseware link you were heading to as a ?next parameter
+        Click "enroll" and be redirected to the page you wanted to go to
+        Click "Don't enroll" and go back to the course about page
+
+    Arguments:
+     - request    : HTTP request
+     - course_id  : course id
+
+    Returns:
+     -RedirectResponse
+    """
+    template_name = 'enroll_staff.html'
+
+    @method_decorator(require_global_staff)
+    @method_decorator(ensure_valid_course_key)
+    def get(self, request, course_id):
+        """
+        Renders Enroll Staff View
+        """
+        user = request.user
+        course_key = from_string_or_404(course_id)
```

Comment:
Here need a  `CourseLocator` that's why I am using this. I have removed the `ensure_valid_course_key `.